bufit.go

```diff
@@ -131,6 +131,9 @@ func (b *Buffer) NextReader() io.ReadCloser {
 // Write appends the given data to the buffer. All active readers will
 // see this write.
 func (b *Buffer) Write(p []byte) (int, error) {
+	if !b.alive() {
+		return 0, io.ErrClosedPipe
+	}
```

Comment:
Here's where things get tricky. What happens if there is a call to Close() after this b.alive() check but before the critical section below?

If we want absolutely no writes to be accepted after Close() is called, then we should lock while closing.